tests/integration_tests_rust.rs

```diff
@@ -315,6 +317,98 @@ fn onchain_spend_receive() {
 	assert!(node_b.list_balances().spendable_onchain_balance_sats < 100000);
 }
 
+#[test]
+fn onchain_wallet_recovery() {
+	let (bitcoind, electrsd) = setup_bitcoind_and_electrsd();
+
+	let chain_source = TestChainSource::Esplora(&electrsd);
+
+	let seed_bytes: Vec<u8> = std::iter::repeat(42u8).take(64).collect();
+
+	let original_config = random_config(true);
+	let original_node = setup_node(&chain_source, original_config, Some(seed_bytes.clone()));
+
+	let premine_amount_sat = 100_000;
+
+	let addr_1 = original_node.onchain_payment().new_address().unwrap();
+
+	premine_and_distribute_funds(
+		&bitcoind.client,
+		&electrsd.client,
+		vec![addr_1],
+		Amount::from_sat(premine_amount_sat),
+	);
+	original_node.sync_wallets().unwrap();
+	assert_eq!(original_node.list_balances().spendable_onchain_balance_sats, premine_amount_sat);
+
+	let addr_2 = original_node.onchain_payment().new_address().unwrap();
+
+	let txid = bitcoind
+		.client
+		.send_to_address(
+			&addr_2,
+			Amount::from_sat(premine_amount_sat),
+			None,
+			None,
+			None,
+			None,
+			None,
+			None,
+		)
+		.unwrap();
+	wait_for_tx(&electrsd.client, txid);
+
+	generate_blocks_and_wait(&bitcoind.client, &electrsd.client, 1);
+
+	original_node.sync_wallets().unwrap();
+	assert_eq!(
+		original_node.list_balances().spendable_onchain_balance_sats,
+		premine_amount_sat * 2
```

Comment:
Can't quite follow this. Is the above a self-send? Or is there a second wallet somewhere that isn't immediately obvious to me?